internal/resources/cluster/data_source_cluster.go

```diff
@@ -96,7 +103,7 @@ func dataSourceTMCClusterRead(ctx context.Context, d *schema.ResourceData, m int
 		_, err = helper.RetryUntilTimeout(getClusterResourceRetryableFn, 10*time.Second, timeoutDuration)
 	}
 
-	if err != nil || resp == nil {
+	if err != nil || resp == nil || resp.Cluster == nil {
```

Comment:
after fixing the above issues check is this extra validation is needed or not

Reply:
This is still needed, else the plugin crashes for the case of apply->remove from TMC (via diff client)->re-apply.